packages/core/theme/src/components/button.ts

```diff
@@ -340,8 +344,45 @@ const button = tv({
     // isInGroup / bordered / ghost
     {
       isInGroup: true,
-      variant: ["bordered", "ghost"],
-      class: "[&:not(:first-child)]:ml-[calc(theme(borderWidth.medium)*-1)]",
+      variant: ["ghost", "bordered"],
+      color: "default",
+      className:
+        "nextui-adjacent-default [&+.nextui-adjacent-default]:ml-[calc(theme(borderWidth.medium)*-1)]",
```

Comment:
I would prefer using `data-color` and `data-variant` over `.nextui-adjacent-…`
```suggestion
        [
          "[[data-variant=bordered][data-color=default]_+_&]:ml-[calc(theme(borderWidth.2)*-1)]",
          "[[data-variant=ghost][data-color=default]_+_&]:ml-[calc(theme(borderWidth.2)*-1)]"
        ],
```

Reply:
You could add a file with all these selectors in `../utils` to make it reusable.
```suggestion
        utils.collapseAdjacentVariantBorders.default,
```

Reply:
When thinking about it using a class name also isn't that bad, it's pretty much just like using `peer`.

Reply:
Something like this would also work.
```suggestion
        "[&+.border-medium.border-default]:ml-[calc(theme(borderWidth.medium)*-1)]",
```

Reply:
Thank you @jguddas 

I have finally placed ```collapseAdjacentVariantBorders``` in the ```utils``` folder to make it reusable. This change also enables support for other border widths in the future.